src/engine/runtime.js

```diff
@@ -451,6 +461,22 @@ class Runtime extends EventEmitter {
         this.emit(Runtime.EXTENSION_ADDED, categoryInfo.blocks.concat(categoryInfo.menus));
     }
 
+    /**
+     * Populate this.monitorBlockInfo
+     */
+    _registerMonitorInfo () {
+        for (const packageName in defaultBlockPackages) {
+            if (defaultBlockPackages.hasOwnProperty(packageName)) {
+                // @todo pass a different runtime depending on package privilege?
+                const packageObject = new (defaultBlockPackages[packageName])(this);
```

Comment:
I think it would be best to only have one instance of each blocks package. It looks like this instance is short-lived, but still I'd rather this function be blended into or called from `_registerBlockPackages` such that the instance used for monitors is the same instance used for other block registration. For built-in blocks this isn't THAT important, but it's pretty important for extensions.